flask_monitor/metrics.py

```diff
@@ -42,19 +31,38 @@ def register_metrics(app, buckets=None, error_fn=None):
     Before CPython 3.6 dictionaries didn't guarantee keys order, so callbacks
     could be executed in arbitrary order.
     """
+
+    if app.config.get("METRICS_ENABLED", False):
+        return app
+    app.config["METRICS_ENABLED"] = True
+    app.logger.info('Metrics enabled')
+
     buckets = [0.1, 0.3, 1.5, 10.5] if buckets is None else buckets
+
+
+    # pylint: disable=invalid-name
+    METRICS_INFO = Gauge(
+        "application_info",
+        "records static application info such as it's semantic version number",
+        ["version"],
+        registry=registry
```

Comment:
@Ziul, from the method signature, `registry` doesn't seem to be a mandatory parameter. If I do not assign value to it, shouldn't we redirect the metrics to `prometheus_client`'s default registry?

Reply:
We could set the registry in the `app.config`, as done with the `METRICS_ENABLED` flag. and always use it from there. In the documentation from `prometheus` they aways create a new registry with `CollectorRegistry()` and there isn't anything about a default registry.

Reply:
@Ziul I'm not sure it's good practice to be putting complex objects in the flask app config, since it should be able to load all configs from a simple config file, for example. 

Given you made the call to `register_metrics` somewhat singleton, we shouldn't bother about storing the registry at the config. Instead, if `registry` is `None` and `config["METRICS_ENABLED"]` is `false`, we should just assign a new `CollectorRegistry()` to the `registry` variable.

Got it?

Reply:
In applications like flask-cache, they use the `app.extensions`  to hold more complicated objects. It's a dictionary like the `app.config` but exclusive to extensions. I can keep the registry instance there.

Reply:
But why you are caching this info, anyway? I don't understand the value it adds to the logic of fixing the problem when `registry==None`.

Reply:
If `register_metrics` and `watch_dependencies` are both called without the registry, they will be stored in a different registry. I'm just trying to avoid this situation, as I believe the intention is to keep the registry as unique for default behavior.

Reply:
@Ziul, alright. Now I see the value. Thank you!